File: Leetcode/2717. Semi-Ordered Permutation.py
# https://leetcode.com/problems/semi-ordered-permutation/

from typing import List


class Solution:
    def semiOrderedPermutation(self, nums: List[int]) -> int:
        f = l = -1
        for i, n in enumerate(nums):
            if n == 1:
                f = i
            elif n == len(nums):
                l = i
        return f + len(nums) - l - 1 - (1 if f > l else 0)

File: Leetcode/2001-3000/2701-2800/2717. Semi-Ordered Permutation.go
// https://leetcode.com/problems/semi-ordered-permutation/

func semiOrderedPermutation(nums []int) int {
    f, l := -1, -1
    for i, n := range nums {
        if n == 1 {
            f = i
        } else if n == len(nums) {
            l = i
        }
    }
    if f > l {
        return f + len(nums) - l - 2
    }
    return f + len(nums) - l - 1
}